### PR TITLE
building metadata with dask

### PR DESCRIPTION
**Pull request by d-sot**
adds support for building metadata with dask 
updates tests

### CODE REVIEW

File: hdf5zarr/hdf5zarr.py [hunk 1]
```diff
@@ -153,7 +166,8 @@ def __init__(self, filename: str, hdf5obj: str = None,
                  collectattrs: bool = True, uri: str = None, hdf5group: str = None,
                  store: Union[MutableMapping, str, Path] = None, store_path: str = None,
                  store_mode: str = 'a', LRU: bool = False, LRU_max_size: int = 2**30,
-                 max_chunksize=2*2**20, driver: str = None, blocksize: int = 2**15, collectrefs: Union[bool, str, list] = None):
+                 max_chunksize=2*2**20, driver: str = None, blocksize: int = 2**15, collectrefs: Union[bool, str, list] = None,
+                 withdask: bool = True, dask_kwds: dict = {}, follow_symlinks: bool = True):
```
Comment (bendichter) [inline hdf5zarr/hdf5zarr.py]: It's better to set default values like `dask_kwds: dict = None`, then later use 

```python
if dask_kwds is not None:
    dask_opts.update(dask_kwds)
```

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

**Comment by bendichter** [comment]
could you explain a little more what you are doing? Some example code would be helpful

File: hdf5zarr/hdf5zarr.py (hunk 1 above)
Comment (d-sot) [reply hdf5zarr/hdf5zarr.py]: https://github.com/catalystneuro/HDF5Zarr/blob/0cc6eb10bfd03b7557e0c8fead785ad1bc88bc5e/hdf5zarr/hdf5zarr.py#L170
https://github.com/catalystneuro/HDF5Zarr/blob/0cc6eb10bfd03b7557e0c8fead785ad1bc88bc5e/hdf5zarr/hdf5zarr.py#L230-L233
good point! It's changed to None now.

**Comment by bendichter** [comment]
@d-sot could you please give more of a description? You are adding a lot of code here and I don't see very many additions to the tests. Are you testing this new Dask feature?

**Comment by d-sot** [comment]
> @d-sot could you please give more of a description? You are adding a lot of code here and I don't see very many additions to the tests. Are you testing this new Dask feature?

Hi, sorry. I didn’t get to this earlier and responding to it now albeit with delay! The pr had updated some tests, and I thought the original fuzz tests were thorough. but I later on I moved on from trying to use dask for this, and instead just with the multiprocessing module, But I got just slightly over 2z performance improvement building the metadata store reading from S3 with multiprocessing compared to the single processing. dask also had some overhead and it didn’t seem like the right choice for this use case.

**Pull request closed without merging**